src/drivers/driver_zephyr.c

```diff
@@ -1546,6 +1596,170 @@ int wpa_drv_zep_deinit_ap(void *priv)
 
 	ret = 0;
 
+out:
+	return ret;
+}
+
+int wpa_drv_zep_sta_add(void *priv, struct hostapd_sta_add_params *params)
+{
+	struct zep_drv_if_ctx *if_ctx = NULL;
+	const struct zep_wpa_supp_dev_ops *dev_ops = NULL;
+	int ret = -1;
+
+	if ((!priv) || (!params)) {
+		wpa_printf(MSG_ERROR, "%s: Invalid params\n", __func__);
+		goto out;
+	}
+
+	if_ctx = priv;
+
+	dev_ops = if_ctx->dev_ctx->config;
+
+	if (!dev_ops->sta_add) {
+		wpa_printf(MSG_ERROR, "%s: sta_add op not supported\n", __func__);
+		goto out;
+	}
+
+	ret = dev_ops->sta_add(if_ctx->dev_priv, params);
+	if (ret) {
+		wpa_printf(MSG_ERROR, "%s: sta_add op failed: %d\n", __func__, ret);
+		goto out;
+	}
+
+	ret = 0;
+
+out:
+	return ret;
+}
+
+int wpa_drv_zep_sta_set_flags(void *priv, const u8 *addr, u32 total_flags, u32 flags_or, u32 flags_and)
+{
+	struct zep_drv_if_ctx *if_ctx = NULL;
+	const struct zep_wpa_supp_dev_ops *dev_ops = NULL;
+	int ret = -1;
+
+	if ((!priv) || (!addr)) {
+		wpa_printf(MSG_ERROR, "%s: Invalid params\n", __func__);
+		goto out;
+	}
+
+	if_ctx = priv;
```

Comment:
Same here, you can set the if_ctx at the beginning of the function without any ill effects.